hyperspy/io_plugins/jeol.py

```diff
@@ -0,0 +1,649 @@
+                            ]["ElevAng"],
+                            "energy_resolution_MnKa": header["PTTD Param"]["Params"][
+                                "PARAMPAGE0_SEM"
+                            ]["MnKaRES"],
+                            "real_time": header["PTTD Data"]["AnalyzableMap MeasData"][
+                                "Doc"
+                            ]["RealTime"],
+                        },
+                    },
+                },
+            },
+            "General": {
+                "original_filename": os.path.basename(filename),
+                "date": datefile.date().isoformat(),
+                "time": datefile.time().isoformat(),
+                "title": "EDX",
+            },
+            "Signal": {
+                "record_by": "spectrum",
+                "quantity": "X-rays (Counts)",
+                "signal_type": "EDS_" + mode,
+            },
+        }
+
+        dictionary = {
+            "data": data,
+            "axes": axes,
+            "metadata": metadata,
+            "original_metadata": header,
+        }
+    else:
+        print("Not a valid JEOL pts format")
+
+    fd.close()
+
+    return dictionary
+
+
+def parsejeol(fd):
+    final_dict = {}
+    tmp_list = []
+    tmp_dict = final_dict
+
+    mark = 1
+    while abs(mark) == 1:
+        mark = np.fromfile(fd, "b", 1)[0]
+        if mark == 1:
+            str_len = np.fromfile(fd, "<i", 1)[0]
+            kwrd = fd.read(str_len).rstrip(b"\x00")
+
+            if (
+                kwrd == b"\xce\xdf\xb0\xc4"
+            ):  # correct variable name which might be 'Port'
+                kwrd = "Port"
+            elif (
+                kwrd[-1] == 222
+            ):  # remove undecodable byte at the end of first ScanSize variable
+                kwrd = kwrd[:-1].decode("utf-8")
+            else:
+                kwrd = kwrd.decode("utf-8")
+            val_type, val_len = np.fromfile(fd, "<i", 2)
+            tmp_list.append(kwrd)
+            if val_type == 0:
+                tmp_dict[kwrd] = {}
+            else:
+                c_type = jTYPE[val_type]
+                arr_len = val_len // np.dtype(c_type).itemsize
+                if c_type == "c":
+                    value = fd.read(val_len).rstrip(b"\x00")
+                    value = value.decode("utf-8").split("\x00")
+                    # value = os.path.normpath(value.replace('\\','/')).split('\x00')
+                else:
+                    value = np.fromfile(fd, c_type, arr_len)
+                if len(value) == 1:
+                    value = value[0]
+                if kwrd[-5:-1] == "PAGE":
+                    kwrd = kwrd + "_" + value
+                    tmp_dict[kwrd] = {}
+                    tmp_list[-1] = kwrd
+                elif kwrd == "CountRate" or kwrd == "DeadTime":
+                    tmp_dict[kwrd] = {}
+                    tmp_dict[kwrd]["value"] = value
+                elif kwrd == "Limits":
+                    print("skip")
```

Comment:
Left over from debugging? Remove this comment or use the logger to print some useful information.

Reply:
Yes I didn't manage to decode the following bytes : 
- first 16 bytes are encode in float32 and looks like limit values ([20. , 1., 2000, 1.] or [1., 0., 1000., 0.001])
- next 4 bytes are ascii character and looks like number format (%.0f or %.3f)
- next 12 bytes are unclear for me
- next 4 bytes are ascii character and are units (kV or nA)
- lest 12 byes are again unclear for me

Reply:
You could  add these comments as comments in the code with the hope it could be useful in the future (for yourself or others). In any, there should not be any `print` statement used here and if you want to display some information for the user, you need to use the [logger](https://docs.python.org/3/library/logging.html) and a suitable level of logging, typical it will `debug`, `info` or `warning`.